blueprints/addon/index.js

```diff
@@ -77,6 +82,21 @@ module.exports = {
     // 100% of addons don't need ember-cli-app-version, make it opt-in instead
     delete contents.devDependencies['ember-cli-app-version'];
 
+    // add scripts to build type declarations for TypeScript addons
+    if (this.options.typescript) {
+      contents.devDependencies.rimraf = '^5.0.1';
+
+      contents.scripts.prepack = 'tsc --project tsconfig.declarations.json';
+      contents.scripts.postpack = 'rimraf declarations';
```

Comment:
Should we also add `declarations/` to `.gitignore`?

Reply:
Yeah, thought about this as well... but then we don't _really_ need that rimraf script anymore, right? The previous behaviour of e-c-ts was to create the declarations only before publishing and immediately delete them afterwards, without them being git-ignored. And this is what is basically happening here as well. 

So should we git-ignore `declarations` and remove rimraf, or have both?

Reply:
Now that the declarations are in their own directory instead of scattered directly throughout the project, I think removing them isn't such a big deal, but I don't feel particularly strongly about it either way 🙂

Reply:
Discussed with embroider core meeting and we recommend adding to gitignore and also calling rimraf either before a fresh build or after (so that extra files are not left around and picked up by typescript).

Reply:
> Should we also add declarations/ to .gitignore?

Just did that!

The `.gitignore` blueprint file is used for both apps and addons, so apps now also get this change. Which isn't really needed. But also that file is already a bit blurry (yarn vs. npm, ember-try stuff for apps), so I think it's ok?